neuland/digitizing/CMakeLists.txt

```diff
@@ -11,25 +11,35 @@
 # or submit itself to any jurisdiction.                                      #
 ##############################################################################
 
-set(INCLUDE_DIRECTORIES 
-${INCLUDE_DIRECTORIES} 
-${R3BROOT_SOURCE_DIR}/neuland/calibration
-${R3BROOT_SOURCE_DIR}/neuland/digitizing
-)
+set(INCLUDE_DIRECTORIES ${INCLUDE_DIRECTORIES} ${R3BROOT_SOURCE_DIR}/neuland/calibration
+                        ${R3BROOT_SOURCE_DIR}/neuland/digitizing)
 include_directories(${INCLUDE_DIRECTORIES})
 
 set(SRCS
-DigitizingEngine.cxx
-DigitizingTacQuila.cxx
-DigitizingTamex.cxx
-R3BNeulandHitMon.cxx
-R3BNeulandDigitizer.cxx
-)
-    
-CHANGE_FILE_EXTENSION(*.cxx *.h HEADERS "${SRCS}")
+    DigitizingChannel.cxx
+    DigitizingPaddle.cxx
+    DigitizingTacQuila.cxx
+    DigitizingTamex.cxx
+    DigitizingPaddleNeuland.cxx
+    R3BNeulandHitMon.cxx
+    R3BNeulandDigitizer.cxx
+    DigitizingPaddleMock.h
+    DigitizingChannelMock.h)
+
+set(HEADERS
+    DigitizingChannel.h
```

Comment:
Hi,

Ok, sorry I didn't know we have this rule. But is this agreement only for the file name or the class name?

But sure I will change the file name first.

Reply:
But does this agreement also apply to the class inside a namespace? Such as 
```cpp
namespace Neuland::Digitizing
{
    class PaddleHit;
}
```
And another question is whether we also have this agreement on the namespace? Because if we only have them on the class name, it would look very funny:
```cpp
Neuland::Digitizing::R3BPaddleHit;
```
Instead I would say, following is much better
```cpp
R3BNeuland::Digitizing::PaddleHit;
```

Reply:
We mostly do not use name spaces because ROOT did not do so in ~1995. 

I would argue that it should be 
```R3B::Neuland::Digitizing::PaddleHit```

Some people would probably argue that there should be a correspondence between the namespace path and the file path. 

I would suggest the following:
* For serializable data, keep the current class naming scheme. 
* For tasks, at least provide an alias:
``
using R3BCalifaMapped2CrystalCal = R3B::Califa::Mapped2Cal;
``
(No idea if this will work well with all the dark magic needed to invoke the tasks within "C scripts". It might be safer to also stick to the legacy naming convention.)
* For implementation details which are not important for people running macros, I think there is no reason not to use naming conventions from our current century instead.

Reply:
For the classes the rule is:
R3B+"Detector name"+"specific names"

For the namespace, please, try to look for the best to ensure that our students can later understand what the code does

Reply:
I would like to discuss a little bit more about `Neuland::Digitizing` and `Digitizing::Neuland`.

If we use `Neuland::Digitizing`, which namespace should we put some common digitizing base classes that can be used in many different detectors? (Like for Neuland and TofD, they both use bar shaped scintillators and similar digitizers.) So normally we put them into a namespace `R3B::Digitizing`. Now if some algorithms from Neuland defined in the `R3B::Neuland::Digitizing` need to use those base classes from `R3B::Digitizing`, they need to go all the way to the top level, like:
```c++
namespace R3B::Neuland::Digitizing
{
    class MyClass : public R3B::Digitizing::Base;
}
```
But on the other hand, if we use `R3B::Digitizing::Neuland`, that becomes very clean:
```c++
namespace R3B::Digitizing::Neuland
{
    class MyClass : public Base;
}
```
With this naming style, we could also reduce some naming chaos like `R3B::Neuland::Digitizing`, `R3B::Tofd::digitizing` or `R3B::Califa::Digi`. Instead, they could just be `R3B::Digitizing::Neuland`, `R3B::Digitizing::Tofd` or `R3B::Digitizing::Califa`.

Reply:
@jose-luis-rs Sure, but is `R3B::Digitizing::Neuland::PaddleHit` ok?

Reply:
> I would like to discuss a little bit more about `Neuland::Digitizing` and `Digitizing::Neuland`.
> 
> If we use `Neuland::Digitizing`, which namespace should we put some common digitizing base classes that can be used in many different detectors? (Like for Neuland and TofD, they both use bar shaped scintillators and similar digitizers.) So normally we put them into a namespace `R3B::Digitizing`. Now if some algorithms from Neuland defined in the `R3B::Neuland::Digitizing` need to use those base classes from `R3B::Digitizing`, they need to go all the way to the top level, like:
> 
> ```c++
> namespace R3B::Neuland::Digitizing
> {
>     class MyClass : public R3B::Digitizing::Base {};
> }
> ```
> 
> But on the other hand, if we use `R3B::Digitizing::Neuland`, that becomes very clean:
> 
> ```c++
> namespace R3B::Digitizing::Neuland
> {
>     class MyClass : public Base {};
> }
> ```
> 
> With this naming style, we could also reduce some naming chaos like `R3B::Neuland::Digitizing`, `R3B::Tofd::digitizing` or `R3B::Califa::Digi`. Instead, they could just be `R3B::Digitizing::Neuland`, `R3B::Digitizing::Tofd` or `R3B::Digitizing::Califa`.

I think that the digitizers for all these detectors are completely different, what is the common part?

Reply:
@jose-luis-rs There are a lot of common parts, like the two side readouts, scintillation, light attenuation, etc. Even if they are very different, they could derived from a shared base class. For example, the simulation of different detectors come from the same base class `R3BDetector`.

Reply:
Yes, but in the case of heavy ions the detection efficiency is around 99% and the "two side readouts, scintillation, light attenuation, etc." are redundant, basically because we only want to determine basic things like the geometrical efficiency.

Of course, what you say is somehow important if one wants to have a real detector response, in particular for the lighter ions. It could be interesting to have a general digitizer class able to manage the digitization of all kinds of particles and provide functions and methods for the other digitizers.